src/components/input-switch/foreignkey-field.tsx

```diff
@@ -83,9 +81,10 @@ const ForeignkeyField = (props: ForeignkeyFieldProps): JSX.Element => {
   const usedFormNumber = typeof props.formNumber === 'number' ? props.formNumber : 1;
 
   const { setValue, getValues } = useFormContext();
+  const { prefillAssociationSelectedRows } = useRecordedit();
 
   const [recordsetModalProps, setRecordsetModalProps] = useState<RecordsetProps | null>(null);
-
+  const [inputSelectedRow, setInputSelectedRow] = useState<SelectedRow | null>(null);
```

Comment:
Why is this state variable needed? We can figure this out by looking at `prefillAssociationSelectedRows`. right? I feel like looking up an array that won't go beyond 200 elements is better than the unnecessary renders that this is going to add.

Reply:
This was left over from when I was updating association selected rows differently. Since it's index based now, you are right this isn't needed and can be a simple lookup each time instead